pkg/exporter/process.go

```diff
@@ -120,39 +116,24 @@ func (ep *ExportingProcess) AddSetAndSendMsg(setType entities.ContentType, set e
 
 	msgBuffer := ep.msg.GetMsgBuffer()
 	var bytesSent int
-	// Check if message is exceeding the limit with new record
-	if uint16(msgBuffer.Len()+ int(ep.set.GetBuffLen())) > entities.MaxTcpSocketMsgSize {
-		ep.set.FinishSet()
-		b, err := ep.sendMsg(set.GetNumberOfRecords())
-		if err != nil {
-			return b, err
-		}
-		bytesSent = bytesSent + b
```

Comment:
We still need to send the message, right? FinishSet would have been called with the previous set.

Reply:
Since we don't keep a `set` field in `exportingProcess` struct and currently don't support multiple set per message, there will be no previous set. For every set passed in the function, if the size is less than `MaxTcpSocketMsgSize`, we will send it because we cannot separate current set into two different sets now. Do you think we should support splitting set or multiple sets per message here?

Reply:
Got it. 
I think split sets case might be important because we do not add any gating in set.AddRecords and add any number of records. I feel it is better to add the gating there and define in the interface that the length of the records should not exceed this `MaxTcpSocketMsgSize`. What do you think?

Let us handle this in a different PRs. Please create new issue for split sets scenario with probable gating in sets. A TODO or an issue for multiple sets would be good.

Reply:
Sounds good. Will create issue then. Thanks for reviewing.